internal/agent/nftables_generator.go

```diff
@@ -31,19 +34,61 @@ var (
 {{ $cfg := . }}
 table {{ .FilterTableType }} {{ .FilterTableName }} {
 	chain {{ .FilterForwardChainName }} {
+		{{- range $dest := $cfg.PolicyAssignments }}
+		ct mark {{ $cfg.FWMarkBits | printf "0x%x" }} and {{ $cfg.FWMarkMask | printf "0x%x" }} ip daddr {{ $dest.Address }} goto pod-{{ $dest.Address }};
+		{{- end }}
 		ct mark {{ $cfg.FWMarkBits | printf "0x%x" }} and {{ $cfg.FWMarkMask | printf "0x%x" }} accept;
 	}
+
+	{{- range $pod := $cfg.PolicyAssignments }}
+	chain pod-{{ $pod.Address }} {
+		{{- range $pol := $pod.NetworkPolicies }}
+		jump {{ $pol }};
+		{{- end }}
+		drop;
+	}
+	{{- end }}
+
+	{{- range $policy := $cfg.NetworkPolicies }}
+	chain {{ $policy.Name }} {
+		{{- range $ruleIndex, $ingressRule := $policy.IngressRuleChains }}
+		jump {{ $policy.Name }}-rule{{ $ruleIndex }};
+		{{- end }}
+	}
+
+	{{- range $ruleIndex, $ingressRule := $policy.IngressRuleChains }}
+	chain {{ $policy.Name }}-rule{{ $ruleIndex }} {
+		{{- range $entryIndex, $entry := $ingressRule.Entries }}
+		{{ $entry.SaddrMatch.Match }} {{ $entry.PortMatch }} {{- if ne ($entry.SaddrMatch.Except | len) 0 }} jump {{ $policy.Name }}-rule{{ $ruleIndex }}-cidr{{ $entryIndex }} {{- else }} accept {{- end }};
+
+		{{- end }}
+	}
```

Comment:
You need to mangle the generated chain names somehow.

Consider the following network policy:

```

apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: test-network-policy
  namespace: k8s-svc-ingress
spec:
  podSelector: {}
  policyTypes:
    - Ingress
  ingress:
    - from:
        - ipBlock:
            cidr: 217.93.63.122/32
        - ipBlock:
            cidr: 172.30.154.0/24
      ports:
        - protocol: TCP
          port: 80
        - protocol: TCP
          port: 443
    - from:
        - namespaceSelector: {}
      ports:
        - protocol: TCP
          port: 80
        - protocol: TCP
          port: 443
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: test-network-policy-rule0
  namespace: k8s-svc-ingress
spec:
  podSelector: {}
  policyTypes:
    - Ingress
  ingress:
    - from:
        - ipBlock:
            cidr: 185.128.119.0/24
            except:
            - 185.128.119.253/32
      ports:
        - protocol: TCP
          port: 80
        - protocol: TCP
          port: 443
```

One way would be to use upper-case letters, which are not allowed in k8s resource names.

Mind to also modify the pod chains accordingly, otherwise a policy called `pod-10.244.157.82` would also be problematic.

Reply:
Oh, that's nasty. 🤦🏽🙈 Good catch!